drivers/memc/Kconfig.sam

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2022 Basalte bv
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
Update Kconfig to new pattern based on autogenerated <BUILD_DIR>/Kconfig/Kconfig.dts.  See https://github.com/zephyrproject-rtos/zephyr/pull/48169 for example of the change.